llvm/lib/Target/Hexagon/HexagonGenExtract.cpp

```diff
@@ -171,7 +171,7 @@ bool HexagonGenExtract::convert(Instruction *In) {
     // this value.
     if (!LogicalSR && (SR > SL))
       return false;
-    APInt A = APInt(BW, ~0ULL).lshr(SR).shl(SL);
+    APInt A = APInt(BW, ~0ULL, true).lshr(SR).shl(SL);
```

Comment:
can we not use APInt::getBitsSet here?

Reply:
I tried this, but it's pretty awkward. Note that the shifts are by different values, so this is not just a simple high bits pattern. Doing this as two shifts directly matches the pattern that's being matched above.